src/main/kotlin/org/javacs/kt/externalsources/ClassInJar.kt

```diff
@@ -10,35 +11,68 @@ import org.javacs.kt.util.KotlinLSException
 import org.javacs.kt.util.replaceExtensionWith
 
 fun parseUriAsClassInJar(uri: String): ClassInJar {
-	val splittedUri = uri.split(".jar!")
-	val jarUri = splittedUri[0] + ".jar"
+	val splitUri = uri.split(".jar!")
+	val jarUri = splitUri[0] + ".jar"
 	val jarPath = Paths.get(URI.create(jarUri))
-	val relativeClassPath = Paths.get(splittedUri[1].trimLeadingPathSeparator())
+	val relativeClassPath = Paths.get(splitUri[1].trimLeadingPathSeparator())
 
-	return ClassInJar(jarPath, relativeClassPath)
+    val relativeJavaPath = Paths.get((splitUri[1].split(".class")[0] + ".java").trimLeadingPathSeparator())
```

Comment:
Theres a good few tab-spaces inconsistencies across all files, not sure which you wanna go with